ops/charms/node_base.py

```diff
@@ -206,6 +223,10 @@ def apply_node_labels(self) -> None:
         # Collect the current label state.
         current_labels = self._stored.current_labels
 
+        juju_az = os.getenv(JUJU_AVAILABILITY_ZONE)
+        if juju_az and not self.get_label(TOPOLOGY_NODE_LABEL):
+            self.set_label(TOPOLOGY_NODE_LABEL, juju_az)
+
```

Comment:
easy peasey.  If this node isn't labeled with the AZ -- label it. 

it doesn't FORCE the node label to something else -- good